### PR TITLE
Migration to fix paragraphs that have 0 length children

### PR DESCRIPTION
**Pull request by shichongrui**
This fixes existing files but another PR will be required to
find the source of these issues and resolve it in the plottr_electron
repository

### CODE REVIEW

File: v2/migrator/migrations/2021.1.4.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+function walkRichContentTree(richContent, callback) {
+  richContent.forEach(o => {
+    callback(o)
+    if (Array.isArray(o.children)) {
+      walkRichContentTree(o.children, callback)
+    }
+  })
+}
+
+function migrate(data) {
```
Comment (cameronsutter) [inline v2/migrator/migrations/2021.1.4.js:29]: normally migrations clone the data as the first step. Did you forget that or do you think it's not necessary? Am I being overly cautious with that?

actually, the first step is to check that the file version isn't this version … but that one I think I am being overly cautious about.

File: v2/migrator/migrations/2021.1.4.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  // nodes
+  data.notes.forEach(note => note.content = fixRCE(note.content))
+
+  // characters
+  const characterParagraphAttributes = [
+    'notes',
+    ...data.customAttributes.characters.filter(attr => attr.type === 'paragraph')
+  ]
+  data.characters.forEach(character => {
```
Comment (cameronsutter) [inline v2/migrator/migrations/2021.1.4.js:41]: characters can also have an array of templates that has an array of attributes and those can be of type 'paragraph' as well

File: v2/migrator/migrations/2021.1.4.js (hunk 1 above)
Comment (shichongrui) [reply v2/migrator/migrations/2021.1.4.js:29]: Nope I just didn't realize that was a necessary step. But thinking about it, it does seem safer.

**Comment by shichongrui** [comment]
I don't know how you test migrations, but considering this touches people's content it seems higher risk and probably should have more testing.

**Comment by cameronsutter** [comment]
We ended up not needing this

**Pull request closed without merging**